scripts/converter/convert_eun_to_nue.py

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env python3
+
+# Copyright 1996-2020 Cyberbotics Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Convert R2020b world file from the EUN (northDirection = '0 0 1') to the NUE coordinate system."""
+
+import math
+import sys
+
+from transforms3d import quaternions
+
+from webots_parser import WebotsParser
+
+
+def rotation(value, r):
+    q0 = quaternions.axangle2quat([float(value[0]), float(value[1]), float(value[2])], float(value[3]))
+    q1 = quaternions.axangle2quat([r[0], r[1], r[2]], r[3])
+    qr = quaternions.qmult(q1, q0)
+    v, theta = quaternions.quat2axangle(qr)
+    return [WebotsParser.str(v[0]), WebotsParser.str(v[1]), WebotsParser.str(v[2]), WebotsParser.str(theta)]
+
+
+def convert_to_nue(filename):
+    world = WebotsParser()
+    world.load(filename)
+
+    for node in world.content['root']:
+        if node['name'] == 'WorldInfo':
+            for field in node['fields']:
+                if field['name'] == 'northDirection':
+                    assert field['value'] == ['0', '0', '1']
+                    # remove the 'northDirection 0 0 1'
+                    del node['fields'][node['fields'].index(field)]
```

Comment:
Would you not need to add a 'coordinate System "NUE"' node since the default is ENU?

Reply:
Yes, this could indeed help, however, if you open a world from R2020a in Webots it will automatically set the coordinate system field to NUE. But that's anyway a good idea to add this filed here if not present.

Reply:
Thank you, it was not clear that the default coordinate system varied based on the webots version in the first line of the world file. I had understood we needed to convert everything to ENU for Webots R2020b+

Reply:
Yes, this mechanism is indeed hidden in Webots, it is there to assure backward compatibility is not broken. In any case, I implemented your suggestion here: #2217